crates/ruff_linter/src/rules/pycodestyle/rules/type_comparison.rs

```diff
@@ -37,90 +39,57 @@ pub struct TypeComparison;
 impl Violation for TypeComparison {
     #[derive_message_formats]
     fn message(&self) -> String {
-        format!("Do not compare types, use `isinstance()`")
+        format!(
+            "Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks"
+        )
     }
 }
 
 /// E721
 pub(crate) fn type_comparison(checker: &mut Checker, compare: &ast::ExprCompare) {
-    for ((left, right), op) in std::iter::once(compare.left.as_ref())
+    for (left, right) in std::iter::once(compare.left.as_ref())
         .chain(compare.comparators.iter())
         .tuple_windows()
         .zip(compare.ops.iter())
+        .filter(|(_, op)| matches!(op, CmpOp::Eq | CmpOp::NotEq))
+        .map(|((left, right), _)| (left, right))
     {
-        if !matches!(op, CmpOp::Is | CmpOp::IsNot | CmpOp::Eq | CmpOp::NotEq) {
-            continue;
+        if is_type(left, checker.semantic()) || is_type(right, checker.semantic()) {
```

Comment:
Unlike pycodestyle, our rule is symmetric... so, e.g., we flag both `type(x) == int` and `int == type(x)`.

Reply:
`is` is commutative, but `==` isn't in Python:
```
[ins] In [14]: (a == b) == (b == a)
Out[14]: False

[ins] In [15]: (a is b) == (b is a)
Out[15]: True
```